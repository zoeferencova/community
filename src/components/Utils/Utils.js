import React from 'react';
import styles from './Utils.module.css';

export function ButtonLight({ className, ...props }) {
    return <button className={`${styles.button} ${className}`} {...props} />
}

export function ButtonDark({ className, ...props }) {
    return <button className={`${styles.buttonDark} ${className}`} {...props} />
}

export function Input({ className, ...props }) {
    return <input className={`${styles.input} ${className}`} {...props} />
}

export function Label({ className, ...props }) {
    return <label className={`${styles.label} ${className}`} {...props} />
}

export function Textarea({ className, ...props }) {
    return <textarea className={`${styles.input} ${className}`} {...props} />
}

export function Select({ className, ...props }) {
    return <select className={`${styles.input} ${className}`} {...props} />
}

export function ProfilePicture({ className, ...props }) {
    let color;
    const colorOptions = ["red", "indigo", "purple", "green", "teal", "blue"]
    const letters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]
    const firstLetter = props.first_name.slice(0, 1);
    const letterIndex = letters.indexOf(firstLetter.toLowerCase());
    color = colorOptions[Math.floor(letterIndex/colorOptions.length)]
    return <div className={`${styles.circle} ${color} ${className}`} {...props}>{firstLetter}</div>
}