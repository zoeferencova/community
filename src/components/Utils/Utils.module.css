.button {
    padding: 8px 16px;
    font-size: 15px;
    line-height: 24px;
    font-family: inherit;
    border: 1px solid rgb(38, 38, 39);
    color: rgb(38, 38, 39);
    border-radius: 3px;
    background-color: transparent;
    transition: 0.2s;
}

.button:hover {
    cursor: pointer;
    opacity: 0.8;
    border: 1px solid rgba(38, 38, 39, 0.8);
}

.buttonDark {
    composes: button;
    background-color: #415166;
    color: white;
    border: 1px solid #415166;
}

.buttonLink:hover {
    border: none;
    background-color: #e4e9f0;
    transition: 0.5s;
    opacity: 1;
}

.input {
    padding: 12px;
    border: 1px solid transparent;
    border-radius: 3px;
    background-color: #f2f5f7;
    display: block;
    margin: 7px auto 25px auto;
    font-family: inherit;
    font-size: 15px;
    width: 100%;
    box-sizing: border-box;
}

.label {
    display: block;
    text-align: left;
    margin: 0 auto;
    font-size: 14px;
    width: 100%;
}

.circle {
    height: 30px;
    width: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 18px;
    color: white;
    font-weight: bold;
    font-size: 14px;
}

:global .red {
    background-color: #bf5656;
}

:global .green {
    background-color: #8cc47a;
}

:global .indigo {
    background-color: #7784bf;
}

:global .purple {
    background-color: #8878b0;
}

:global .teal {
    background-color: #78b09a
}

:global .blue {
    background-color: #56bcbf;
}