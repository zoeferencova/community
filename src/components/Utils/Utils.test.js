import React from "react";
import { shallow } from "enzyme";
import { ButtonLight, ButtonDark, Input, Label, Textarea, ProfilePicture, Select, Error } from "./Utils";


describe("ButtonLight", () => {
    it("renders ButtonLight component with button text when not loading", () => {
        const wrapper = shallow(<ButtonLight loading="false">Hello</ButtonLight>);

        expect(wrapper).toMatchSnapshot();
    });

    it("renders ButtonLight component with loader icon when loading", () => {
        const wrapper = shallow(<ButtonLight loading="true">Hello</ButtonLight>);

        expect(wrapper).toMatchSnapshot();
    });
});

describe("ButtonDark", () => {
    it("renders ButtonDark component with button text when not loading", () => {
        const wrapper = shallow(<ButtonDark loading="false">Hello</ButtonDark>);

        expect(wrapper).toMatchSnapshot();
    });

    it("renders ButtonDark component with loader icon when loading", () => {
        const wrapper = shallow(<ButtonDark loading="true">Hello</ButtonDark>);

        expect(wrapper).toMatchSnapshot();
    });
});

describe("Input", () => {
    it("renders Input", () => {
        const wrapper = shallow(<Input />);

        expect(wrapper).toMatchSnapshot();
    });
});

describe("Label", () => {
    it("renders Label", () => {
        const wrapper = shallow(<Label />);

        expect(wrapper).toMatchSnapshot();
    });
});

describe("Textarea", () => {
    it("renders Textarea", () => {
        const wrapper = shallow(<Textarea />);

        expect(wrapper).toMatchSnapshot();
    });
});

describe("ProfilePicture", () => {
    const mockProps = {
        first_name: "Zoe"
    }

    it("renders ProfilePicture", () => {
        const wrapper = shallow(<ProfilePicture {...mockProps} />);

        expect(wrapper).toMatchSnapshot();
    });
});

describe("Select", () => {
    it("renders Select", () => {
        const wrapper = shallow(<Select />);

        expect(wrapper).toMatchSnapshot();
    });
});

describe("Error", () => {
    const mockProps = {
        message: "Error"
    }

    it("renders Error", () => {
        const wrapper = shallow(<Error {...mockProps} />);

        expect(wrapper).toMatchSnapshot();
    });
});



