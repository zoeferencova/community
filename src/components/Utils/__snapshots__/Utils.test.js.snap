// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDark renders ButtonDark component with button text when not loading 1`] = `
<button
  className="buttonDark "
  loading="false"
>
  Hello
</button>
`;

exports[`ButtonDark renders ButtonDark component with loader icon when loading 1`] = `
<button
  className="buttonDark "
  loading="true"
>
  <i
    className="fa fa-spinner fa-spin spinner"
  />
</button>
`;

exports[`ButtonLight renders ButtonLight component with button text when not loading 1`] = `
<button
  className="button "
  loading="false"
>
  Hello
</button>
`;

exports[`ButtonLight renders ButtonLight component with loader icon when loading 1`] = `
<button
  className="button "
  loading="true"
>
  <i
    className="fa fa-spinner fa-spin spinner"
  />
</button>
`;

exports[`Error renders Error 1`] = `
<div
  className="error "
  message="Error"
  role="alert"
>
  <i
    className="fas fa-exclamation-circle"
  />
   
  Error
</div>
`;

exports[`Input renders Input 1`] = `
<input
  className="input "
/>
`;

exports[`Label renders Label 1`] = `
<label
  className="label "
/>
`;

exports[`ProfilePicture renders ProfilePicture 1`] = `
<div
  className="circle blue "
  first_name="Zoe"
>
  Z
</div>
`;

exports[`Select renders Select 1`] = `
<select
  className="input "
/>
`;

exports[`Textarea renders Textarea 1`] = `
<textarea
  className="input "
/>
`;
